src/LinAlg/matrix.h

```diff
@@ -963,19 +964,21 @@ namespace utl //! General utility classes and functions.
 
   template<> inline
   vector<float>& vector<float>::add(const std::vector<float>& X,
-                                    const float& alfa)
+                                    const float& alfa,
+                                    int ofsx, int stridex)
   {
     size_t n = this->size() < X.size() ? this->size() : X.size();
-    cblas_saxpy(n,alfa,&X.front(),1,this->ptr(),1);
+    cblas_saxpy(n,alfa,&X[ofsx],stridex,this->ptr(),1);
```

Comment:
Or isn't it better with `X.data()+ofsx` in this case?

Reply:
sure.